### PR TITLE
Enhance the docker-compose/makefile setup for running/reverting migrations

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## This PR contains the following changes

- Added discrete setup/migrate/rollback commands to makefile which trigger corresponding docker-compose services.
- Updated root migration to include a rollback step that drops everthing

`make db-setup` - Run the database migrations + seeding
`make db-migrate` - Run the database migrations
`make db-rollback` - Rollback the latest migration(s)

Note:
`make web` - still runs `db-setup` as part of its workflow, so if you're just building the app for regular use then you don't need to worry about any of the above commands.

Note:
If the root migration gets rolled-back, then the migrations AND seeding will have been reverted.  So getting the app back to its previous state would need a `make db-setup`, not just a `make db-migrate`.

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,19 +94,22 @@ build-backend: ## Builds all backend containers
 	@echo "==============================================="
 	@echo "Make: build-backend - building backend images"
 	@echo "==============================================="
-	@docker-compose -f docker-compose.yml build db db_setup api nginx clamav
+	@docker-compose -f docker-compose.yml build db db_setup api nginx
+## @docker-compose -f docker-compose.yml build db db_setup api nginx clamav
```
Comment (sdevalapurkar) [inline Makefile:98]: can this commented line be removed?

Comment (sdevalapurkar) [reply Makefile:98]: same question for few more places below

**Review by sdevalapurkar — COMMENTED** [review]
nice! just a question about few commented lines

File: Makefile (hunk 1 above)
Comment (NickPhura) [reply Makefile:98]: I'm just commenting out the clamav step, but didnt want to lose track of it, as we will eventually set that up on our file upload endpoints

**Review by sdevalapurkar — APPROVED** [review]

**Comment by anissa-agahchen** [comment]
Looks good.  I tested it locally and it works great!

**Review by anissa-agahchen — APPROVED** [review]

**Pull request merged**